JavaFiddle-ejb/src/main/java/ru/javafiddle/core/ejb/CompileAndRunBean.java

```diff
@@ -1,27 +1,259 @@
+import java.lang.reflect.Modifier;
+import java.lang.reflect.Type;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
 
 /**
  * Created by artyom on 15.12.15.
  */
 @Stateless
 @Named(value = "compileAndRunBean")
-public class CompileAndRunBean {
+public class CompileAndRunBean extends DynamicCompiler {
 
-    @PersistenceContext
+    @PersistenceContext(name = "JFPersistenceUnit")
     EntityManager em;
 
+    @EJB
+    FileBean fileBean;
+
+
+    public static final String DEFAULT_PACKAGE_PREFFIX = "";
+    private static final Logger LOG = Logger.getLogger(CompileAndRunBean.class);
+
+    public enum FileType {
+        CLASS(8),
+        INTERFACE(9),
+        EXCEPTION(10),
+        RUNNABLE(11),
+        ENUM(12),
+        ANNOTATION(13);
+
+        private int type;
+
+        FileType(int type) {
+            this.type = type;
+        }
+
+        public int getType() {
+            return type;
+        }
+
+        public static boolean isFile(int type) {
+            for (FileType fileType : FileType.values()) {
+                if (fileType.getType() == type) {
+                    return true;
+                }
+            }
+            return false;
+        }
+    }
+
+    private final int SOURCES = 3;
+
+
+
+    public void init(ClassLoader parentLoader) throws Exception {
+        super.init(parentLoader);
+    }
+
     //!TODO
-    public String compile(String projectHash) {
-        return null;
+    public String compile(String projectHash) throws Exception {
+//        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+//        PrintStream printStream = new PrintStream(baos);
+//        PrintStream old = System.out;
+//        System.setOut(printStream);
+        BasicConfigurator.configure();
+        List<File> sources;
+        sources = fileBean.getProjectFiles(projectHash);
+        init(ClassLoader.getSystemClassLoader());
+        List<SimpleJavaFileObject> userSources = constructResources(sources);
+        String message = "";
+        try {
+            message = compileToClass(userSources);
+        } catch (Exception e) {
+            LOG.error("Exception, can't compile", e);
+            e.printStackTrace();
+        }
+        LOG.info(message);
+//        System.out.flush();
+//        System.setOut(old);
+        return message;
     }
 
     //!TODO
     public String run(String projectHash) {
-        return null;
+        List<File> sources;
+        sources = fileBean.getProjectFiles(projectHash);
+        String mes = "";
+        if (!StringUtils.isEmpty(getMessage())) {
+            try {
+                createClasses(sources, projectHash);
+                String mainMeth = findMain(sources, getClassLoader());
+                Process process = null;
+                java.io.File proj = new java.io.File(projectHash + "/src");
+                process = Runtime.getRuntime().exec("java " + mainMeth, null, proj);
```

Comment:
ProcessBuilder is more recommended API due to the following reasons:
1) You control the start of the process and can do all preparations (in/out stream handling, env setup)
2) You can setup the env for your process (different from your main one) - recommended to do.

Next, where do you wait for your subprocess to finish? Looks like nowhere...
That means you won't be able to handle the output in case of a relatively large project. I.e. you will show weird results to the user initiated the compile&run.